In [2]:
import pandas as pd
import numpy as np

In [3]:
df=pd.read_csv("titles.csv")

In [116]:
df.head(1)

id                                title  type  \
0  ts300399  Five Came Back: The Reference Films  SHOW   

                                         description  release_year  \
0  This collection includes 12 World War II-era p...          1945   

  age_certification  runtime             genres production_countries  seasons  \
0             TV-MA       51  ['documentation']               ['US']      1.0   

  imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
0     NaN         NaN         NaN              0.6         NaN

In [22]:
df.index

RangeIndex(start=0, stop=5850, step=1)

### Total no of movies =5850

In [11]:
df.columns

Index(['id', 'title', 'type', 'description', 'release_year',
       'age_certification', 'runtime', 'genres', 'production_countries',
       'seasons', 'imdb_id', 'imdb_score', 'imdb_votes', 'tmdb_popularity',
       'tmdb_score'],
      dtype='object')

### Total number of columns = 15

In [13]:
df.isnull().sum()

id                         0
title                      1
type                       0
description               18
release_year               0
age_certification       2619
runtime                    0
genres                     0
production_countries       0
seasons                 3744
imdb_id                  403
imdb_score               482
imdb_votes               498
tmdb_popularity           91
tmdb_score               311
dtype: int64

### Seasons is the column with larger number of missing values followed by age_certifications


In [106]:
df[df["seasons"]==1.0]["id"].count() 

np.int64(1221)

### 1221 SHOWS has one season

In [107]:
df[df["seasons"]==2.0]["id"].count()

np.int64(389)

### 389 TV SHOWS has two seasons

In [108]:
df[df["seasons"]==3.0]["id"].count()

np.int64(187)

### 187 SHOWS has three seasons

In [109]:
df[df["seasons"]==4.0]["id"].count()

np.int64(120)

### 120 shows has 4 seasons

In [110]:
df[df["seasons"]>4.0]["id"].count()

np.int64(189)

In [160]:
df["seasons"].value_counts()

seasons
1.0     1221
2.0      389
3.0      187
4.0      120
5.0       79
6.0       36
7.0       18
8.0       11
9.0        9
11.0       8
10.0       6
12.0       4
15.0       3
24.0       2
13.0       2
25.0       1
39.0       1
42.0       1
14.0       1
21.0       1
37.0       1
29.0       1
19.0       1
32.0       1
20.0       1
16.0       1
Name: count, dtype: int64

### 110 has more than 4 seasons

In [114]:
max = df["seasons"].max()
df[df["seasons"]==max]

id     title  type  \
91  ts20981  Survivor  SHOW   

                                          description  release_year  \
91  A reality show contest where sixteen or more c...          2000   

   age_certification  runtime       genres production_countries  seasons  \
91             TV-PG       48  ['reality']               ['US']     42.0   

      imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
91  tt0239195         7.4     24687.0            54.19         7.4

### SURVIVOR has hightest number of seasons 

In [167]:
seasons = df.groupby("seasons")["imdb_score"].mean()
seasons

seasons
1.0     6.871679
2.0     6.939189
3.0     7.227072
4.0     6.987288
5.0     7.412658
6.0     7.448571
7.0     7.577778
8.0     7.200000
9.0     8.222222
10.0    6.950000
11.0    7.650000
12.0    7.100000
13.0    6.750000
14.0    3.800000
15.0    7.633333
16.0    7.300000
19.0    7.600000
20.0    7.800000
21.0    8.800000
24.0    6.050000
25.0    7.400000
29.0    6.500000
32.0    8.700000
37.0    7.800000
39.0    6.700000
42.0    7.400000
Name: imdb_score, dtype: float64

In [168]:
seasons.max()

np.float64(8.8)

### most liked duration of shows is 21 seasons

In [23]:
df.groupby("type")["id"].count()

type
MOVIE    3744
SHOW     2106
Name: id, dtype: int64

### Total number of movies= 3744
### Total number of shows= 2106

In [117]:
df['release_year'].value_counts()

release_year
2019    836
2020    814
2021    787
2018    773
2017    563
       ... 
1971      1
1977      1
1958      1
1978      1
1963      1
Name: count, Length: 63, dtype: int64

In [46]:
df['release_year'].value_counts().idxmax()

np.int64(2019)

### 2019 is largest producing year

In [47]:
df['release_year'].value_counts().idxmin()

np.int64(1985)

### 1985 is least producing year

In [4]:
movie=df[df["type"]=="MOVIE"]
show =df[df["type"]=="SHOW"]

In [85]:
movie["release_year"].value_counts().idxmax()

np.int64(2019)

In [87]:
movie["release_year"].value_counts().idxmin()

np.int64(1985)

### 1985 is least movies producing year

### 2019 is largest movies producing year

In [86]:
show["release_year"].value_counts().idxmax()

np.int64(2020)

In [89]:
show["release_year"].value_counts().idxmin()

np.int64(1988)

### 1988 is least shows producing year

### 2020 is largest shows producing year

In [68]:
movie["runtime"].mean()

np.float64(98.21367521367522)

### average runtime of movies is 98 imnutes

In [8]:
long= movie["runtime"].idxmax()
movie.loc[long
# only fetch single row in case of ties

id                                                               tm469911
title                                                      Bonnie & Clyde
type                                                                MOVIE
description             Based on the true story of Clyde Barrow, a cha...
release_year                                                         2013
age_certification                                                   PG-13
runtime                                                               240
genres                                     ['drama', 'crime', 'european']
production_countries                                               ['AT']
seasons                                                               NaN
imdb_id                                                         tt2374851
imdb_score                                                            NaN
imdb_votes                                                            NaN
tmdb_popularity                       

In [13]:
long = movie["runtime"].max()
movie[movie["runtime"]==long]

id           title   type  \
923  tm469911  Bonnie & Clyde  MOVIE   

                                           description  release_year  \
923  Based on the true story of Clyde Barrow, a cha...          2013   

    age_certification  runtime                          genres  \
923             PG-13      240  ['drama', 'crime', 'european']   

    production_countries  seasons    imdb_id  imdb_score  imdb_votes  \
923               ['AT']      NaN  tt2374851         NaN         NaN   

     tmdb_popularity  tmdb_score  
923            0.933         7.5

### Bonnie &Cycle is the longest movie having 240 minutes released in 2013

In [14]:
short =movie["runtime"].min()
movie[movie["runtime"]==short]

id          title   type  \
5766  tm1004011  Time to Dance  MOVIE   

                                            description  release_year  \
5766  When a ballroom dancer’s shot at a crucial tou...          2021   

     age_certification  runtime                genres production_countries  \
5766               NaN        2  ['romance', 'drama']               ['IN']   

      seasons    imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
5766      NaN  tt8622232         2.2       963.0            1.043         4.0

### Time to Dance is the shortest movie having 2 minutes released in 2021

In [69]:
show["runtime"].mean()

np.float64(38.978157644824314)

### average runtime of shows is 38 minutes

In [15]:
longer=show["runtime"].max()
show[show["runtime"]==longer]

id                title  type  \
515  ts19384  A Lion in the House  SHOW   

                                           description  release_year  \
515  Five families struggle with the ups and downs ...          2006   

    age_certification  runtime             genres production_countries  \
515             TV-14      190  ['documentation']               ['US']   

     seasons imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
515      1.0     NaN         NaN         NaN            1.487         7.5

### A Lion in the House is the longest shows released in 2006(190 min)

In [17]:
shorter= show["runtime"].min()
show[show["runtime"]==shorter]["title"]

518                       Pecados capitales
594                               Pink Zone
1340                                La Niña
1430                  Sons of the Caliphate
2577                                 Borges
3887                     Emily's Wonder Lab
4394                     Nailed It! Germany
4680                          นางสาวก้นครัว
5152                 Bullsh*t the Game Show
5223                             Back to 15
5390             Storybots Laugh Learn Sing
5498    Vendetta: Truth, Lies and The Mafia
5515                         Pedal to Metal
5714             Word Party Presents: Math!
Name: title, dtype: object

In [18]:
show[show["runtime"]==shorter]["title"].count()

np.int64(14)

### 14 shows having least runtime of 0 minutes

In [120]:
movie[movie["runtime"]<90]["id"].count()

np.int64(1163)

In [121]:
movie[movie["runtime"]>150]["id"].count()

np.int64(162)

In [106]:
high= movie["imdb_score"].max()
movie[movie["imdb_score"]==high]

id                              title   type  \
1092  tm250247  Chhota Bheem & Krishna vs Zimbara  MOVIE   
4771  tm996762                              Major  MOVIE   

                                            description  release_year  \
1092  Chhota Bheem Aur Krishna vs Zimbara is an Indi...          2013   
4771  Based on the life of real-life Hero Major Sand...          2022   

     age_certification  runtime               genres production_countries  \
1092                 G       64        ['animation']                   []   
4771               NaN      149  ['action', 'drama']               ['IN']   

      seasons    imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  \
1092      NaN  tt6417984         9.1        16.0            1.442         NaN   
4771      NaN  tt9851854         9.1      7576.0            6.429       8.188   

     length_category  
1092           Short  
4771          Medium

In [20]:
low=movie["imdb_score"].min()
movie[movie["imdb_score"]==9.1]

id                              title   type  \
1092  tm250247  Chhota Bheem & Krishna vs Zimbara  MOVIE   
4771  tm996762                              Major  MOVIE   

                                            description  release_year  \
1092  Chhota Bheem Aur Krishna vs Zimbara is an Indi...          2013   
4771  Based on the life of real-life Hero Major Sand...          2022   

     age_certification  runtime               genres production_countries  \
1092                 G       64        ['animation']                   []   
4771               NaN      149  ['action', 'drama']               ['IN']   

      seasons    imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
1092      NaN  tt6417984         9.1        16.0            1.442         NaN  
4771      NaN  tt9851854         9.1      7576.0            6.429       8.188

In [26]:
#movie[movie['imdb_score'] == movie['imdb_score'].max()]


In [28]:
hightest=show["imdb_score"].max()
show[show["imdb_score"]==hightest]

id     title  type  \
2658  ts265844  #ABtalks  SHOW   

                                            description  release_year  \
2658  #ABtalks is a YouTube interview show hosted by...          2018   

     age_certification  runtime genres production_countries  seasons  \
2658             TV-PG       68     []                   []      1.0   

         imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
2658  tt12635254         9.6         7.0              NaN         NaN

### #ABtalks(released in 2018} has hightest imdb ratings of 9.6

In [30]:
lowest=show["imdb_score"].min()
show[show["imdb_score"]==lowest]

id                              title  type  \
5298  ts280693  Thomas & Friends: All Engines Go!  SHOW   
5695  ts319303                     He's Expecting  SHOW   

                                            description  release_year  \
5298  In a reboot of the classic TV series, a younge...          2021   
5695  When a successful ad executive who's got it al...          2022   

     age_certification  runtime  \
5298              TV-Y       11   
5695             TV-PG       26   

                                                 genres production_countries  \
5298  ['animation', 'fantasy', 'drama', 'family', 'm...               ['US']   
5695          ['drama', 'comedy', 'romance', 'fantasy']               ['JP']   

      seasons     imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  
5298      1.0  tt13940396         2.0       555.0            1.287         8.0  
5695      1.0  tt14719276         2.0      3019.0           11.314         4.0

### Thomas & Friends: All Engines Go! and He's Expecting has a ties with 2.0 rating

In [38]:
df["genres"]

0                                 ['documentation']
1                                ['drama', 'crime']
2       ['drama', 'action', 'thriller', 'european']
3                   ['fantasy', 'action', 'comedy']
4                                 ['war', 'action']
                           ...                     
5845                           ['romance', 'drama']
5846                                      ['drama']
5847                                     ['comedy']
5848                                             []
5849              ['family', 'animation', 'comedy']
Name: genres, Length: 5850, dtype: object

In [44]:
count = df["genres"].value_counts()
count

genres
['comedy']                                            484
['documentation']                                     329
['drama']                                             328
['comedy', 'drama']                                   135
['drama', 'romance']                                  124
                                                     ... 
['action', 'drama', 'fantasy', 'scifi']                 1
['family', 'drama', 'comedy', 'sport']                  1
['romance', 'comedy', 'crime', 'drama']                 1
['comedy', 'crime', 'action', 'drama', 'thriller']      1
['drama', 'thriller', 'history']                        1
Name: count, Length: 1726, dtype: int64

In [43]:
count.idxmax()

"['comedy']"

In [46]:
#df["genres"].value_counts().idxmax()

### COMEDY IS MOST COMMON GENRE

In [ ]:
# max_val = df['genres'].max()
# second_max_value = df[df['genres'] < max_val]['genres'].max()
# second_max_value


In [47]:
df["genres"].value_counts().idxmin()

"['documentation', 'music', 'reality']"

### LEAST COMMON GENRES ARE DOCUMENTATION ,REALITY AND MUSIC

In [62]:
high = df.groupby("genres")["imdb_score"].max()
high

genres
['action', 'animation', 'comedy', 'drama', 'family', 'fantasy', 'music', 'romance']    7.7
['action', 'animation', 'comedy', 'drama', 'family', 'scifi']                          7.9
['action', 'animation', 'comedy', 'family', 'fantasy']                                 7.8
['action', 'animation', 'comedy', 'family', 'music']                                   6.0
['action', 'animation', 'comedy', 'family']                                            7.6
                                                                                      ... 
['western', 'history', 'drama']                                                        6.5
['western', 'horror', 'action']                                                        6.0
['western', 'thriller', 'horror']                                                      5.5
['western']                                                                            NaN
[]                                                                                 

In [57]:
high.idxmax()

'[]'

In [63]:
high.sort_values(ascending=False)

genres
[]                                                          9.6
['reality']                                                 9.5
['drama', 'crime', 'thriller']                              9.5
['scifi', 'animation', 'action', 'family', 'fantasy']       9.3
['documentation']                                           9.3
                                                           ... 
['scifi', 'documentation']                                  NaN
['thriller', 'comedy', 'romance']                           NaN
['thriller', 'horror', 'animation', 'drama', 'european']    NaN
['thriller', 'horror', 'european']                          NaN
['western']                                                 NaN
Name: imdb_score, Length: 1726, dtype: float64

### MOST POPULAR genre is reality

In [65]:
least = df.groupby("genres")["imdb_score"].min()
least.idxmin()

"['thriller', 'drama', 'scifi']"

In [68]:
least.sort_values()

genres
['thriller', 'drama', 'scifi']                              1.5
['comedy', 'family', 'european']                            1.6
['comedy', 'drama', 'action']                               1.7
['comedy']                                                  1.9
['comedy', 'family']                                        2.0
                                                           ... 
['scifi', 'documentation']                                  NaN
['thriller', 'comedy', 'romance']                           NaN
['thriller', 'horror', 'animation', 'drama', 'european']    NaN
['thriller', 'horror', 'european']                          NaN
['western']                                                 NaN
Name: imdb_score, Length: 1726, dtype: float64

### LEAST POPULAR genres are thriller,drama and science fiction

In [70]:
movie["genres"].value_counts().idxmax()

"['comedy']"

In [74]:
movie["genres"].value_counts().sort_values(ascending =False)

genres
['comedy']                                           384
['documentation']                                    230
['drama']                                            224
['comedy', 'documentation']                          100
['comedy', 'drama']                                   84
                                                    ... 
['action', 'thriller', 'crime', 'sport']               1
['animation', 'fantasy', 'family', 'comedy']           1
['drama', 'horror', 'music', 'thriller', 'crime']      1
['thriller', 'drama', 'horror', 'scifi']               1
['thriller', 'family', 'horror', 'comedy']             1
Name: count, Length: 1105, dtype: int64

In [75]:
movie["genres"].value_counts().sort_values()

genres
['drama', 'romance', 'thriller', 'horror']             1
['action', 'crime', 'comedy']                          1
['action', 'thriller', 'crime', 'sport']               1
['animation', 'fantasy', 'family', 'comedy']           1
['drama', 'horror', 'music', 'thriller', 'crime']      1
                                                    ... 
['comedy', 'drama']                                   84
['comedy', 'documentation']                          100
['drama']                                            224
['documentation']                                    230
['comedy']                                           384
Name: count, Length: 1105, dtype: int64

In [71]:
movie["genres"].value_counts().idxmin()

"['animation', 'family', 'scifi']"

In [72]:
show["genres"].value_counts().idxmax()

"['reality']"

In [73]:
show["genres"].value_counts().idxmin()

"['animation', 'comedy', 'family', 'music', 'fantasy', 'european']"

In [78]:
more_popular=movie.groupby("genres")["imdb_score"].max()

In [79]:
more_popular.sort_values(ascending =False)

genres
['animation']                                               9.1
['action', 'drama']                                         9.1
['documentation']                                           8.9
['drama']                                                   8.9
['drama', 'romance']                                        8.8
                                                           ... 
['fantasy', 'comedy', 'animation']                          NaN
['fantasy', 'scifi']                                        NaN
['thriller', 'comedy', 'romance']                           NaN
['thriller', 'horror', 'animation', 'drama', 'european']    NaN
['thriller', 'horror', 'european']                          NaN
Name: imdb_score, Length: 1105, dtype: float64

In [80]:
least_popular=movie.groupby("genres")["imdb_score"].min()
least_popular.sort_values()

genres
['thriller', 'drama', 'scifi']                              1.5
['comedy', 'family', 'european']                            1.6
['comedy', 'drama', 'action']                               1.7
['comedy']                                                  1.9
['comedy', 'family']                                        2.0
                                                           ... 
['fantasy', 'comedy', 'animation']                          NaN
['fantasy', 'scifi']                                        NaN
['thriller', 'comedy', 'romance']                           NaN
['thriller', 'horror', 'animation', 'drama', 'european']    NaN
['thriller', 'horror', 'european']                          NaN
Name: imdb_score, Length: 1105, dtype: float64

In [82]:
more_popular=show.groupby("genres")["imdb_score"].max()
more_popular.sort_values(ascending=False)

genres
[]                                                       9.6
['reality']                                              9.5
['drama', 'crime', 'thriller']                           9.5
['documentation']                                        9.3
['scifi', 'animation', 'action', 'family', 'fantasy']    9.3
                                                        ... 
['scifi', 'documentation']                               NaN
['scifi', 'thriller', 'action']                          NaN
['thriller']                                             NaN
['war']                                                  NaN
['western']                                              NaN
Name: imdb_score, Length: 846, dtype: float64

In [83]:
least_popular=show.groupby("genres")["imdb_score"].min()
least_popular.sort_values()

genres
['drama', 'comedy', 'romance', 'fantasy']                         2.0
['animation', 'fantasy', 'drama', 'family', 'music', 'comedy']    2.0
['reality']                                                       2.1
['drama', 'horror']                                               2.3
['reality', 'comedy']                                             2.4
                                                                 ... 
['scifi', 'documentation']                                        NaN
['scifi', 'thriller', 'action']                                   NaN
['thriller']                                                      NaN
['war']                                                           NaN
['western']                                                       NaN
Name: imdb_score, Length: 846, dtype: float64

In [84]:
df["production_countries"].value_counts()

production_countries
['US']                1959
['IN']                 599
['JP']                 264
[]                     229
['KR']                 222
                      ... 
['ES', 'PE']             1
['BE', 'GB', 'DE']       1
['ZA', 'GB', 'US']       1
['CL', 'PE']             1
['CA', 'CN', 'US']       1
Name: count, Length: 452, dtype: int64

In [90]:
df["production_countries"].value_counts().idxmax()

"['US']"

In [95]:
data=df["production_countries"].value_counts().sort_values(ascending = False)
data

production_countries
['US']                1959
['IN']                 599
['JP']                 264
[]                     229
['KR']                 222
                      ... 
['ES', 'PE']             1
['BE', 'GB', 'DE']       1
['ZA', 'GB', 'US']       1
['CL', 'PE']             1
['CA', 'CN', 'US']       1
Name: count, Length: 452, dtype: int64

In [97]:
data.head(6)

production_countries
['US']    1959
['IN']     599
['JP']     264
[]         229
['KR']     222
['GB']     218
Name: count, dtype: int64

In [99]:
movie["production_countries"].value_counts().idxmax()

"['US']"

In [102]:
df2=movie["production_countries"].value_counts().sort_values(ascending=False)
df2.head(6)

production_countries
['US']    1184
['IN']     555
[]         161
['JP']      99
['ES']      92
['GB']      87
Name: count, dtype: int64

In [100]:
show["production_countries"].value_counts().idxmax()

"['US']"

In [103]:
df2=show["production_countries"].value_counts().sort_values(ascending=False)
df2.head(6)

production_countries
['US']    775
['KR']    178
['JP']    165
['GB']    131
['ES']     69
[]         68
Name: count, dtype: int64

In [109]:
movie["length_category"].value_counts()

length_category
Medium    2419
Short     1163
Long       162
Name: count, dtype: int64

In [117]:
df['length'] = df['runtime'].apply(lambda x: 'Long' if x > 150 else('short' if x<90 else 'medium'))
df.head(2)

id                                title   type  \
0  ts300399  Five Came Back: The Reference Films   SHOW   
1   tm84618                          Taxi Driver  MOVIE   

                                         description  release_year  \
0  This collection includes 12 World War II-era p...          1945   
1  A mentally unstable Vietnam War veteran works ...          1976   

  age_certification  runtime              genres production_countries  \
0             TV-MA       51   ['documentation']               ['US']   
1                 R      114  ['drama', 'crime']               ['US']   

   seasons    imdb_id  imdb_score  imdb_votes  tmdb_popularity  tmdb_score  \
0      1.0        NaN         NaN         NaN            0.600         NaN   
1      NaN  tt0075314         8.2    808582.0           40.965       8.179   

   length  
0   short  
1  medium

In [118]:
df.groupby('length')['imdb_score'].mean()

length
Long      6.829487
medium    6.224989
short     6.725252
Name: imdb_score, dtype: float64

In [120]:
df.groupby('length')['imdb_votes'].mean()

length
Long      75620.487179
medium    32256.995710
short     13423.986736
Name: imdb_votes, dtype: float64

### On basis of imdb rating , people prefer long movies 
### On basis of imdb voting, people prefer short movies

In [123]:
max_votes=movie["imdb_votes"].max()
movie[movie["imdb_votes"]==max_votes]

id      title   type  \
237  tm92641  Inception  MOVIE   

                                           description  release_year  \
237  Cobb, a skilled thief who commits corporate es...          2010   

    age_certification  runtime                                    genres  \
237             PG-13      148  ['action', 'scifi', 'music', 'thriller']   

    production_countries  seasons    imdb_id  imdb_score  imdb_votes  \
237         ['US', 'GB']      NaN  tt1375666         8.8   2294231.0   

     tmdb_popularity  tmdb_score length_category  
237          108.284         8.4          Medium

### Inception is most voted movie on IMDB

In [125]:
min_votes=movie["imdb_votes"].min()
movie[movie["imdb_votes"]==min_votes]

id               title   type  \
2798   tm407349  The Birth Reborn 2  MOVIE   
4470   tm835414              Sitara  MOVIE   
4564  tm1041549           Dear Affy  MOVIE   

                                            description  release_year  \
2798                                                NaN          2018   
4470  A Bangaldeshi housewife crosses the border to ...          2019   
4564  An engaged couple hit a number of speed bumps ...          2020   

     age_certification  runtime                  genres production_countries  \
2798                PG       91       ['documentation']               ['BR']   
4470               NaN      128  ['drama', 'animation']               ['IN']   
4564               NaN      126   ['romance', 'comedy']               ['NG']   

      seasons     imdb_id  imdb_score  imdb_votes  tmdb_popularity  \
2798      NaN  tt16114198         6.6         5.0            1.400   
4470      NaN   tt7654668         8.0         5.0            1.115   
4564      NaN  tt14928724         5.2         5.0            0.939   

      tmdb_score length_category  
2798         5.8          Medium  
4470         7.1          Medium  
4564         NaN          Medium

### The Birth Reborn 2 , Sitara	and Dear Affy is least voted movie on IMDB

In [126]:
max_votes=show["imdb_votes"].max()
show[show["imdb_votes"]==max_votes]

id         title  type  \
216  ts4  Breaking Bad  SHOW   

                                           description  release_year  \
216  When Walter White, a New Mexico chemistry teac...          2008   

    age_certification  runtime                          genres  \
216             TV-MA       48  ['drama', 'crime', 'thriller']   

    production_countries  seasons    imdb_id  imdb_score  imdb_votes  \
216               ['US']      5.0  tt0903747         9.5   1775990.0   

     tmdb_popularity  tmdb_score  
216          353.848        8.79

In [127]:
min_votes=movie["imdb_votes"].min()
show[show["imdb_votes"]==min_votes]

id                        title  type  \
5257  ts312094  JAPAN SINKS: People of Hope  SHOW   
5670  ts296581             The Mystic River  SHOW   
5830  ts288861                    Let's Eat  SHOW   

                                            description  release_year  \
5257  In 2023 in Tokyo, Prime Minister Eiichi Higash...          2021   
5670  As pregnant women vanish in a remote Nigerian ...          2021   
5830  A food-blogging insurance agent encounters a f...          2021   

     age_certification  runtime                     genres  \
5257             TV-14       55  ['scifi', 'war', 'drama']   
5670             TV-14       40                 ['horror']   
5830             TV-14       60        ['drama', 'comedy']   

     production_countries  seasons     imdb_id  imdb_score  imdb_votes  \
5257               ['JP']      1.0  tt15569106         6.0         5.0   
5670               ['US']      1.0  tt15134080         5.8         5.0   
5830               ['TH']      1.0  tt15978704         8.2         5.0   

      tmdb_popularity  tmdb_score  
5257            7.557         6.5  
5670            1.441         NaN  
5830            2.334         NaN

### Breaking Bed is most voted show on IMDB.
### JAPAN SINKS , The Mystic River and Let's Eat is least voted show on IMDB

In [130]:
df['age_certification'].value_counts()

age_certification
TV-MA    883
R        556
TV-14    474
PG-13    451
PG       233
TV-PG    188
G        124
TV-Y7    120
TV-Y     107
TV-G      79
NC-17     16
Name: count, dtype: int64

### MOST COMMON MATURITY RATING : TV-MA(ABOVE 17 YEARS)

In [131]:
df2= pd.read_csv("credits.csv")

In [132]:
df2.head(3)

person_id       id            name      character   role
0       3748  tm84618  Robert De Niro  Travis Bickle  ACTOR
1      14658  tm84618    Jodie Foster  Iris Steensma  ACTOR
2       7064  tm84618   Albert Brooks            Tom  ACTOR

In [142]:
data =pd.merge(df ,df2 ,on="id")

In [157]:
data["name"].value_counts().sort_values(ascending =False)

name
Kareena Kapoor Khan    25
Boman Irani            25
Shah Rukh Khan         23
Takahiro Sakurai       21
Amitabh Bachchan       20
                       ..
Adelaida Buscato        1
Luz Stella Luengas      1
Inés Prieto             1
Isabel Gaona            1
Jodie Foster            1
Name: count, Length: 54314, dtype: int64

### Kareena Kapoor Khan is the  most frequent actress.
### Boman Irani is the most frequent actor.

In [146]:
data["character"].value_counts()

character
Self                         1950
Himself                      1276
Herself                       426
Self (archive footage)        330
Dancer                        149
                             ... 
Peichia                         1
Chinchin's Father               1
Personell Officer               1
Melio, Delicatessen Owner       1
Mafioso (uncredited)            1
Name: count, Length: 47274, dtype: int64